akka-http2-support/src/main/scala/akka/http/impl/engine/http2/FrameEvent.scala

```diff
@@ -4,37 +4,62 @@
 
 package akka.http.impl.engine.http2
 
-import akka.http.impl.engine.http2.Protocol.FrameType
-import akka.http.impl.engine.http2.Protocol.SettingIdentifier
+import akka.http.impl.engine.http2.Http2Protocol.FrameType
+import akka.http.impl.engine.http2.Http2Protocol.SettingIdentifier
 import akka.util.ByteString
 
-sealed trait FrameEvent
+sealed trait FrameEvent {
+  def flags: ByteFlag
+  def hasFlag(f: ByteFlag): Boolean = f.isSet(flags)
+}
 sealed trait StreamFrameEvent extends FrameEvent {
   def streamId: Int
 }
 
 final case class DataFrame(
-  streamId:  Int,
-  endStream: Boolean,
-  payload:   ByteString) extends StreamFrameEvent
+  flags:    ByteFlag,
+  streamId: Int,
+  payload:  ByteString) extends StreamFrameEvent {
+  def endStream = Http2Protocol.Flags.END_STREAM.isSet(flags)
+}
 
 final case class HeadersFrame(
+  flags:               ByteFlag,
```

Comment:
Hmm, good points. I also have mixed feelings, esp. if not all frames use headers - this makes the current version from this PR a big ugly. Removing from model classes but using in rendering sounds good, I'll address that somewhere during my day today 👍